### ISSUE TITLE
xbox one controller

### ISSUE THREAD

**Issue by asp0909** [Ryochan7/DS4Windows#366]
will you implement the emulation of the control driver from xbox one to DS4?

**Comment by Ryochan7** [comment]
Probably never. I don't think I would get around to working on a mapper for non-DS4 controllers. Also, XInput input reading would have to be supported in order to read from the Xbox One controller as opposed to reading an HID device like what is done for reading DS4 controllers. I think InputMapper 1.7 is supposed to support that functionality so you can try it and see if you have any luck with that program.

https://beta.inputmapper.com/download/inputmapper-1-7-nightly-build/

A couple of related problems would be finalizing ViGEmBus support and getting DS4 emulation to work. ViGEmBus support is pretty much complete but I am waiting for a new signed driver to get released before DS4Windows will use it by default; an alt build is available for people wanting to use ViGEmBus support now. ViGEmBus is able to emulate a DS4 wired controller so it would just be a matter of having some setting in DS4Windows specifying which controller type should be emulated.

**Comment by Ryusennin** [comment]
Maybe I'm wrong, but I believe asp0909 was asking if DS4Windows would support the XOne controller API (if there's any difference at all with the X360).

**Comment by Ryochan7** [comment]
I assumed that asp0909 was trying to use an Xbox One controller with PS4
Remote Play. ViGEmBus supports emulating an Xbox One controller although I
don't know when someone would use that option rather than emulating an Xbox
360 controller. Does the Xbox streaming app require using an Xbox One
controller?

**Comment by Ryochan7** [comment]
Luckily I have taken the DS4WindowsAPI approach and split the recent DS4Library code so I have a more minimal base to use for testing. I have tested emulating a DS4 using ViGEmBus and I got it to work. It took some time to get performance to an acceptable level but it just took a few minor tweaks. Using a DS4 v1 connected to the Sony Wireless Adapter yielded better performance in Rocket League compared to using the same controller connected via USB. Rocket League is the only game in my library that I know for sure has native DS4 support.
